src/SignalR/docs/specs/TransportProtocols.md

```diff
@@ -186,6 +186,32 @@ Long Polling requires that the client poll the server for new messages. Unlike t
 
 A Poll is established by sending an HTTP GET request to `[endpoint-base]` with the following query string parameters
 
+## HTTP Streaming (HTTP/1, HTTP/2, HTTP/3)
+
+HTTP Streaming uses a single bi-directional HTTP request or 2 uni-directional HTTP requests as a single connection (one for input, one for output). If the HTTP client in use supports bi-directional streaming, negotiation can be skipped (if the client knows the server supports the streaming transport), otherwise, it requires a connection already be established using the `POST [endpoint-base]/negotiate` request.
+
+### 1 bi-directional HTTP Request
+
+For clients that support bi-directional streaming, the HTTP Streaming transport is activated by making a POST request `[endpoint-base]` with the content type `application/bedrock-streaming`. The **optional** `id` query string value is used to identify the connection to attach to. 
+
+If there is no `id` query string value, a new connection is established. If the parameter is specified but there is no connection with the specified ID value, a `404 Not Found` response is returned. 
+
+Establishing a second streaming connection when there is already a connection associated with the Endpoints connection is not permitted and will fail with a `409 Conflict` status code.
+
+Errors while establishing the connection are handled by returning a `500 Server Error` status code as the response to the upgrade request. This includes errors initializing EndPoint types. 
+
+### 1 uni-directional HTTP Requests (Server-to-Client only)
+
+For clients that do not support bi-directional streaming (like browser clients), it requires a connection already be established using the POST `[endpoint-base]/negotiate` request. The HTTP Streaming transport running in this mode is activated by making a GET request `[endpoint-base]` with the content type `application/bedrock-streaming`.
```

Comment:
How does the server know to allow POSTs because this is the uni-directional version?

Reply:
The POST acts as the upload half of the stream in this case? How does the client know the stream was successfully established?

Reply:
Presumably by getting a 200 response from the "GET request `[endpoint-base]` with the content type `application/bedrock-streaming`". By the time the server responds with the 200 to a get request, the client should be free to POST since at that point the server would not about the client and its unidirectionalness.

Reply:
For SSE (which is very similar) I believe browsers wait until the headers arrive. I expect we'd do the same here (assuming there's some way to observe that in the browser). If not, we may need some kind of "hello" message to be sent down the stream to confirm that it's working.

Reply:
Having tested this with fetch and streaming there was a need for a "hello" message only on firefox (which was also needed in SSE only on firefox)